code/modules/mob/living/carbon/brain/MMI.dm

```diff
@@ -18,14 +18,21 @@
 /obj/item/device/mmi/update_icon()
 	if(brain)
 		if(istype(brain,/obj/item/organ/internal/brain/alien))
-			icon_state = "mmi_alien"
+			if(brain.brainmob && brain.brainmob.stat == DEAD)
+				icon_state = "mmi_alien_dead"
+			else
+				icon_state = "mmi_alien"
 			braintype = "Xenoborg" //HISS....Beep.
 		else
-			icon_state = "mmi_full"
+			if(brain.brainmob && brain.brainmob.stat == DEAD)
+				icon_state = "mmi_dead"
+			else
+				icon_state = "mmi_full"

```

Comment:
I moved the mmi's death icon_state update in update_icon() where it belongs. Instead of being manually changed in death().